### PR TITLE
Split torch ops into separate files

### LINKED ISSUE #29: Move op methods into dedicated objects and export them to torch namespace
`torch.scala` is getting a bit large, and we still need to implement a bunch of ops. So I think it makes sense to move the op methods into smaller units and then define aliases via export clauses in the `torch` namespace/package like we already do with `torch.special`.

The difference being that we should do it also for methods that are not aliased explicitly in PyTorch (methods that are defined in the `torch` namespace there). A natural way to split them, is to use the [ops structure in the PyTorch docs](https://pytorch.org/docs/stable/torch.html#math-operations). We already group into these op groups in `torch.scala` already so it shouldn't be hard to move things out.

The advantage should be more clarity, maintainability and smaller compilation units.

Here's an idea for a structure:

`torch/ops/PointwiseOps.scala`
```scala
package torch.ops
object PointwiseOps:
  def abs[D <: NumericNN](input: Tensor[D]): Tensor[D] = ...
```
`torch/ops/ReductionOps.scala`
```scala
package torch.ops
object ReductionOps:
  def argmax[D <: DType]( ...
```
`torch/torch.scala`
```scala
export ops.PointwiseOps.*
export ops.ReductionOps.*
```

I just did a quick feasibility test and there seems an issue with default methods with default parameters which are lost when an export is imported in a different compilation unit. See https://github.com/lampepfl/dotty/issues/17930. I could not find a way around it. Since we're we're using default parameters extensively, this is a blocker and I guess we'll have to wait until it is fixed.

### PR DESCRIPTION
**Pull request by davoclavo**
Fixes #29

Bonus additions:

- Add `torch.multinomial` [RandomSamplingOps]
- Add `torch.randint` [RandomSamplingOps]
- Add `torch.gradient` [PointwiseOps]
- Add `torch.nn.functional.oneHot`

### CODE REVIEW

**Comment by sbrunk** [comment]
Thanks so much @davoclavo! No worries, I'll just rebase and resolve any conflicts.
